File: seed/csharp-model/examples/src/SeedExamples/Types/Directory.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using SeedExamples.Core;

namespace SeedExamples;

public record Directory
{
    [JsonPropertyName("name")]
    public required string Name { get; set; }

    [JsonPropertyName("files")]
    public IEnumerable<File>? Files { get; set; }

    [JsonPropertyName("directories")]
    public IEnumerable<Directory>? Directories { get; set; }

    /// <summary>
    /// Additional properties received from the response, if any.
    /// </summary>
    /// <remarks>
    /// [EXPERIMENTAL] This API is experimental and may change in future releases.
    /// </remarks>
    [JsonExtensionData]
    public IDictionary<string, JsonElement> AdditionalProperties { get; internal set; } =
        new Dictionary<string, JsonElement>();

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/ruby-sdk-v2/objects-with-imports/lib/seed/file/directory/directory.rb

module Seed
    module Types
        class Directory < Internal::Types::Model
            field :name, String, optional: false, nullable: false
            field :files, Internal::Types::Array[Seed::file::File], optional: true, nullable: false
            field :directories, Internal::Types::Array[Seed::file::directory::Directory], optional: true, nullable: false

    end
end
